resolvers.py

```diff
@@ -0,0 +1,335 @@
+from analytics.graphql.constants import *
+from tools.gql.resolvers import DjangoListResolver
+from analytics import models
+from django.db.models import Sum, F, Q
+from django.db.models.expressions import Window
+from django.db.models.functions import RowNumber
+
+
+class GenericResolver(DjangoListResolver):
+    @classmethod
+    def preprocess_queryset(
+        cls,
+        qs,
+        source,
+        info,
+        sr_fields=None,
+        pr_fields=None,
+        model=None,
+        filter_options=None,
+        search_options=None,
+        order_options=None,
+        *args,
+        **kwargs,
+    ):
+
+        return qs
+
+
+class AccountWisePodDetailsResolver:
+    def __call__(self, source, info, *args, **kwargs):
+        try:
+
+            def stringify(data_item):
+                # To avoid problems of serialization converting all types to str
+                if type(data_item) not in [type("char")]:  # , type(1)]:
+                    # To avoid null values to reach FE overriding with 0
+                    if data_item is None:
+                        data_item = 0
+                    return str(data_item)
+                else:
+                    return data_item
+
+            # function to get current period info
+            def get_current_period():
+                current_period_info = models.PeriodsFacts.objects.filter(
+                    period_tag__iexact="current_period"
+                )
+                return current_period_info.values()[0]
+
+            account_name_received = kwargs.get(
+                "account_name", DEFAULT_ACCOUNT_TO_DISPLAY
+            )
+            # if NBI is selected get the details of hardcode podnames else the details of specific account name
+            if account_name_received == DEFAULT_ACCOUNT_TO_DISPLAY:
+                qs = models.MasterPodDetailsFact.objects.filter(pod_name__in=NBI_PODS)
+            else:
+                qs = models.MasterPodDetailsFact.objects.filter(
+                    account_name__iexact=account_name_received
+                )
+            # taking sample object to group the keys for year wise demarcation
+            try:
+                sample = qs.values()[0]
+            except Exception as e:
+                raise Exception("Account Name(s) not found ..!" + str(e))
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Using `Exception` and `BaseException` can make your code prone to errors and difficult to maintain. Instead, we recommend using one of the [Built-in Exceptions](https://docs.python.org/3/library/exceptions.html) or creating a custom exception class that is derived from `Exception` or one of its subclasses.